xdsl/dialects/builtin.py

```diff
@@ -1598,29 +1597,28 @@ def get_strides(self) -> Sequence[int | None] | None:
             case _:
                 return self.layout.get_strides()
 
-    @classmethod
-    def constr(
-        cls,
-        *,
-        shape: GenericAttrConstraint[Attribute] | None = None,
-        element_type: GenericAttrConstraint[_MemRefTypeElement] = AnyAttr(),
-        layout: GenericAttrConstraint[Attribute] | None = None,
-        memory_space: GenericAttrConstraint[Attribute] | None = None,
-    ) -> GenericAttrConstraint[MemRefType[_MemRefTypeElement]]:
-        if (
-            shape is None
-            and element_type == AnyAttr()
-            and layout is None
-            and memory_space is None
-        ):
-            return BaseAttr[MemRefType[_MemRefTypeElement]](MemRefType)
-        return ParamAttrConstraint[MemRefType[_MemRefTypeElement]](
-            MemRefType, (shape, element_type, layout, memory_space)
-        )
+
+def MemRefTypeConstr(
+    *,
+    shape: GenericAttrConstraint[Attribute] | None = None,
+    element_type: GenericAttrConstraint[_MemRefTypeElement] = AnyAttr(),
+    layout: GenericAttrConstraint[Attribute] | None = None,
+    memory_space: GenericAttrConstraint[Attribute] | None = None,
+) -> GenericAttrConstraint[MemRefType[_MemRefTypeElement]]:
+    if (
+        shape is None
+        and element_type == AnyAttr()
+        and layout is None
+        and memory_space is None
+    ):
+        return BaseAttr[MemRefType[_MemRefTypeElement]](MemRefType)
+    return ParamAttrConstraint[MemRefType[_MemRefTypeElement]](
+        MemRefType, (shape, element_type, layout, memory_space)
+    )
 
 
 AnyMemRefType: TypeAlias = MemRefType[Attribute]
-AnyMemRefTypeConstr = BaseAttr[MemRefType[Attribute]](MemRefType)
```

Comment:
I would personally prefer to remove extra aliases. We already have quite a few ways to build these constraints.

Reply:
I just feel unhappy about constructing these objects many times, but that may well be premature optimisation...

Reply:
This is surely an immeasurable difference